frontend/src/metabase/lib/entity-id/hooks/use-validated-entity-id.ts

```diff
@@ -0,0 +1,77 @@
+import { skipToken } from "metabase/api";
+import { useTranslateEntityIdQuery } from "metabase/api/entity-id";
+import type { BaseEntityId, CardId, DashboardId } from "metabase-types/api";
+import { isBaseEntityID } from "metabase-types/api/entity-id";
+
+type SUPPORTED_ENTITIES = { dashboard: DashboardId; card: CardId };
+
+/**
+ * A hook that validates and potentially translates an entity ID.
+ *
+ * If the hook is loading, then `null` is returned.
+ * If the ID is a valid NanoID and the translation is successful, the translated ID is returned.
+ * Otherwise, the original ID is returned.
+ *
+ * @param {Object} params - The parameters for the hook.
+ * @param {EntityType} params.type - The type of the entity (e.g., 'card', 'dashboard', etc.).
+ * @param {string | number | null | undefined} params.id - The ID to validate and potentially translate.
+ *
+ */
+export const useValidatedEntityId = <
```

Comment:
This names makes me think this hook an id and returns a "validated entity id", I know it doesn't sound pretty but I'd call it something like `useMaybeTranslateEntityId` since that's kinda what it does, it **maybe** translated an entity id to an id